pkg/operator/operator.go

```diff
@@ -133,6 +137,34 @@ func New(ctx context.Context, config operatorconfig.Config, kubeConfig *rest.Con
 		return nil, fmt.Errorf("failed to create status controller: %v", err)
 	}
 
+	// Set up the DNSNameResolver controller. This controller is unmanaged by
+	// the manager; the enable_dnsnameresolver controller starts it and the
+	// caches.
+	dnsNameResolverController, dnsNameResolverControllerCaches, err := dnsnameresolver.NewUnmanaged(operatorManager, dnsnameresolver.Config{
+		OperandNamespace: operatorcontroller.DefaultOperandNamespace,
+		ServiceName: operatorcontroller.DNSServiceName(&operatorv1.DNS{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: operatorcontroller.DefaultDNSController,
+			},
+		}).Name,
+		DNSPort:                  fmt.Sprint(manifests.DNSDaemonSet().Spec.Template.Spec.Containers[0].Ports[0].ContainerPort),
+		DNSNameResolverNamespace: operatorcontroller.DefaultDNSNameResolverNamespace,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create dnsnameresolver controller: %v", err)
+	}
+
+	// Set up the enable_dnsnameresolver controller.
```

Comment:
Is this what actually starts the DNSNameResolver controller?  Please mention in comments.